Created html and view for editlab_forinstructors

BMHebert opened the pull request:
Created an html page and view for the page that will allow instructors to edit the assignments of TAs to particular lab sections.

BMHebert committed:
Created html and view for editlab_forinstructors

Created an html page and view for the page that will allow instructors to edit the assignments of TAs to particular lab sections.

BMHebert commented:
The change to home.html is just a temporary link to the page for testing and demonstration purposes. It can be easily removed at a later time.

YazanSalem reviewed (commented):
Changes look good to me!

BMHebert committed:
Implemented schedule functionality

Removed schedule model since it didn't really do anything and would actually cause more problems if we kept it. I believe, after experimenting with the code, that it is better to dynamically generate the schedule when it is needed than keep them in the database, this way we do not have to update the schedule every time a change is made to the database.

The way I got it to work was by making a method for the UserProfile class called getCourseLabList which returns a list of all the courses and labs associated with a particular user, sorted by start time. I also made a method for Course and Lab called getDays which parses the string listing all the days of the course and returns a list of all of them. This makes it much easier to work with. To make sure all input is formatted correctly, I updated the html for anything that could change the day of a lab or course so that instead of manual text entry, they have to choose from a selection of days.

Finally I updated the html and view for schedule to display the schedule of the current user given that they are an instructor or ta. Currently it does nothing for supervisors since they do not have schedules.

BMHebert commented:
Big update. Read more in the most recent commit.

BMHebert committed:
Merge branch 'main' into BMHebert

BMHebert committed:
Update urls.py

Tiny error fix

BMHebert committed:
Bug fixes

Fixed a few small things causing errors.

BMHebert committed:
Merge branch 'BMHebert' of https://github.com/YazanSalem/CS361Project into BMHebert

BMHebert committed:
Update editcourse.html

Small bug fix

BMHebert committed:
Updated edit html files

Added in checks so that in editlab and editcourse for and selections the user has to make, the default options are already selected.
